portal/views/user.py

```diff
@@ -1642,7 +1642,7 @@ def posted_filename(req):
             abort(
                 400, "no file found - please POST a single file using "
                      "standard multipart/form-data parameters")
-        key = req.files.keys()[0]  # either 'file' or actual filename
+        key = list(req.files.keys())[0]  # either 'file' or actual filename
```

Comment:
What I found is that `.keys()` returns an iterator in py3 (instead of a list in py2), so wrapping `req.files.keys()` in `list()` forces it into a list so that the `[0]` works in py3

Reply:
Thanks for the detailed explanation, that makes perfect sense :)
In that case, try using [`next()`](https://stackoverflow.com/questions/4741243/how-to-pick-just-one-item-from-a-generator-in-python/4741269#4741269). that way the whole generator doesn't need to be evaluated:
```
key = next(req.files.keys())
```

Reply:
Gotcha, that works

Reply:
Looks like it's erroring in python2 now, maybe try this?:
```
key = next(iter(req.files.keys()))

```

Reply:
Good catch, just saw that too. I'll try that.